test/regression/builder/road_curve_factory_test.cc

```diff
@@ -462,6 +474,54 @@ TEST_F(RoadCurveFactoryMakeLaneWidthTest, MultipleLaneWidth) {
   EXPECT_NEAR(kWidthDotAtP1, dut->f_dot(kP1), kLinearTolerance);
 }
 
+TEST_F(RoadCurveFactoryMakeLaneWidthTest, EnsureContiguityNoAdaptThrows) {
+  EXPECT_THROW(
+      road_curve_factory_->MakeLaneWidth(kLaneWidthsToAdapt, kP0, kP1, kEnsureContiguity, kDontAdaptLaneWidths),
+      maliput::common::assertion_error);
+}
+
+class RoadCurveFactoryAdaptLaneWidthTest : public ::testing::Test {
+ protected:
+  const double kAngularTolerance{1e-10};
+  const double kLinearTolerance{1e-10};
+  const double kScaleLength{1.};
+  const double kAdaptedCubicPolyLength{1e-1};
+  void SetUp() override {
+    road_curve_factory_ = std::make_unique<RoadCurveFactory>(kLinearTolerance, kScaleLength, kAngularTolerance);
+  }
+  std::unique_ptr<RoadCurveFactory> road_curve_factory_;
+};
+
+TEST_F(RoadCurveFactoryAdaptLaneWidthTest, AdaptConstantLaneWidths) {
+  xodr::LaneWidth a{0. /* offset */, .002 /* a */, 0. /* b */, 0. /* c */, 0. /* d */};
+  xodr::LaneWidth b{2. /* offset */, .001 /* a */, 0. /* b */, 0. /* c */, 0. /* d */};
+  xodr::LaneWidth adapted = road_curve_factory_->AdaptCubicPolynomial(a, b);
+  EXPECT_NEAR(adapted.s_0, b.s_0 - kAdaptedCubicPolyLength, kLinearTolerance);
+  EXPECT_NEAR(adapted.a, .002, kLinearTolerance);
+  EXPECT_NEAR(adapted.b, 0., kLinearTolerance);
+  EXPECT_NEAR(adapted.c, -.30, kLinearTolerance);
+  EXPECT_NEAR(adapted.d, 2.0, kLinearTolerance);
```

Comment:
Also evaluate value and derivative to guarantee this is doing its job and adapting the stuff

Reply:
Added a different test for this.